table/block_based/block_based_table_reader.cc

```diff
@@ -2078,8 +2086,10 @@ Status BlockBasedTable::GetDataBlockFromCache(
     GetContext* get_context) const {
   BlockCacheLookupContext lookup_data_block_context(
       TableReaderCaller::kUserMultiGet);
+  assert(block_type == BlockType::kData);
```

Comment:
This function could be called on index too, right?

Reply:
That's the MultiGet-specific GetDataBlockFromCache overload. It is only called for data blocks; the only call site is https://github.com/facebook/rocksdb/blob/master/table/block_based/block_based_table_reader.cc#L3445 .